### PR TITLE
added issues/assign endpoint

### PR DESCRIPTION
**Pull request by kknqqq**
# PR Details

Added issues/assign endpoint to provide possibility to assign issues automatically to old runs and results

## Related Issue

https://jira.itransition.com/browse/RNDTESTAUTO-128

## How Has This Been Tested

I had a test run with element that was not in the state exist error
image
I've added an issue with regex
image
I've sent a request to endpoint
image
The issues were assigned:
image
Same was done for the testrun with javascript error in it:
image
image
image
image

I've also added an error handling if there's nothing to assign to (no failed results) or the IDs are wrong:
image

And Updated Swagger:
image

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/main/controllers/Project/ResultController.java [hunk 1]
```diff
@@ -167,4 +179,28 @@ private void fillResultSteps(TestResultDto result) throws AqualityException {
         stepResultTemplate.setProject_id(result.getProject_id());
         result.setSteps(stepResultController.get(stepResultTemplate));
     }
+
+    public Map<String, Integer> matchIssues(Integer testResultId) throws AqualityException {
+        TestResultDto testResultTemplate = new TestResultDto();
+        testResultTemplate.setId(testResultId);
+        List<TestResultDto> testResults = this.get(testResultTemplate);
+        testResults = testResults.stream().filter(x -> x.getFinal_result_id() != 2 && x.getFail_reason() != null && x.getIssue_id() == null).collect(Collectors.toList());
```
Comment (kirylRunets) [inline src/main/java/main/controllers/Project/ResultController.java:187]: add constant for '2' value or reuse existing so it won't look like a magic number or even create a function to get filtered list

File: src/main/java/main/controllers/Project/ResultController.java [hunk 2]
```diff
@@ -167,4 +179,28 @@ private void fillResultSteps(TestResultDto result) throws AqualityException {
...
+        if (testResults.isEmpty()) {
+            throw new AqualityException("No test result found to update. Wrong ID might be provided.");
+        }
+        IssueDto issueTemplate = new IssueDto();
+        issueTemplate.setProject_id(testResults.get(0).getProject_id());
+        List<IssueDto> issues = issueController.get(issueTemplate);
+        Integer count = 0;
+        for (IssueDto issue : issues) {
```
Comment (kirylRunets) [inline src/main/java/main/controllers/Project/ResultController.java:195]: create a function for the loop

File: src/main/java/main/controllers/Project/TestRunController.java [hunk 3]
```diff
@@ -156,10 +161,46 @@ private List<TestRunDto> fillMilestonesAndSuites(List<TestRunDto> testRuns) thro
...
+                    .findFirst().orElse(null));
+            testRun.setTest_suite(
+                    suites.stream().filter(x -> x.getId().equals(testRun.getTest_suite_id())).findFirst().orElse(null));
+            testRun.setLabel(
+                    labels.stream().filter(x -> x.getId().equals(testRun.getLabel_id())).findFirst().orElse(null));
         }
         return testRuns;
     }
+
+    public Map<String, Integer> matchIssues(Integer testRunId) throws AqualityException {
```
Comment (kirylRunets) [inline src/main/java/main/controllers/Project/TestRunController.java:174]: same comments, split the function into smalller pieces

File: src/main/java/main/view/Project/IssuesAssignServlet.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse resp) {
+        setPostResponseHeaders(resp);
+        setEncoding(resp);
+        try {
+            Session session = createSession(req);
+            Map<String, Integer> results = new HashMap<>();
+            Integer testRunId = getIntegerQueryParameter(req, "testRunId");
+            Integer testResultId = getIntegerQueryParameter(req, "testResultId");
+            if (testRunId != null) {
```
Comment (kirylRunets) [inline src/main/java/main/view/Project/IssuesAssignServlet.java:28]: what if I specify both testRunId and testResultId?

Comment (kknqqq) [reply src/main/java/main/view/Project/IssuesAssignServlet.java:28]: it will use testRunID and ignore the testResultId. Should it be changed?

**Comment by kknqqq** [comment]
Resolved the comments. Screenshots of updated functionality:
image
image
image
image
image

**Review by kirylRunets — APPROVED** [review]

**Pull request merged**